azext_iot/iothub/providers/message_endpoint.py

```diff
@@ -0,0 +1,351 @@
+    ):
+        super(MessageEndpoint, self).__init__(cmd, hub_name, rg, dataplane=False)
+        self.support_cosmos = hasattr(self.hub_resource.properties.routing.endpoints, "cosmos_db_sql_collections")
+        self.cli = EmbeddedCLI(cli_ctx=self.cmd.cli_ctx)
+
+    def create(
+        self,
+        endpoint_name: str,
+        endpoint_type: str,
+        endpoint_account_name: Optional[str] = None,
+        endpoint_resource_group: Optional[str] = None,
+        endpoint_subscription_id: Optional[str] = None,
+        endpoint_policy_name: Optional[str] = None,
+        connection_string: Optional[str] = None,
+        container_name: Optional[str] = None,
+        encoding: Optional[str] = None,
+        batch_frequency: int = 300,
+        chunk_size_window: int = 300,
+        file_name_format: str = '{iothub}/{partition}/{YYYY}/{MM}/{DD}/{HH}/{mm}',
+        endpoint_uri: Optional[str] = None,
+        entity_path: Optional[str] = None,
+        database_name: Optional[str] = None,
+        primary_key: Optional[str] = None,
+        secondary_key: Optional[str] = None,
+        partition_key_name: Optional[str] = None,
+        partition_key_template: Optional[str] = None,
+        identity: Optional[str] = None
+    ):
+        resource_group_name = self.hub_resource.additional_properties["resourcegroup"]
+        if not endpoint_resource_group:
+            endpoint_resource_group = resource_group_name
+        if not endpoint_subscription_id:
+            endpoint_subscription_id = self.hub_resource.additional_properties['subscriptionid']
+
+        authentication_type = AuthenticationType.KeyBased.value
+        endpoint_identity = None
+        if identity:
+            authentication_type = AuthenticationType.IdentityBased.value
+            if identity != SYSTEM_ASSIGNED_IDENTITY:
+                endpoint_identity = ManagedIdentity(
+                    user_assigned_identity=identity
+                )
+        elif not connection_string:
+            # check for args to get the connection string
+            error_msg = "Please provide a connection string '--connection-string/-c'"
+            if not (
+                endpoint_account_name and entity_path and endpoint_policy_name
+            ) and endpoint_type.lower() in [
+                EndpointType.EventHub.value, EndpointType.ServiceBusQueue.value, EndpointType.ServiceBusTopic.value
+            ]:
+                raise ArgumentUsageError(
+                    error_msg + " or endpoint namespace '--endpoint-namespace', endpoint "
+                    "entity path '--entity-path', and policy name '--policy-name'."
+                )
+            elif not endpoint_account_name and endpoint_type.lower() in [
+                EndpointType.AzureStorageContainer.value, EndpointType.CosmosDBContainer.value
+            ]:
+                raise ArgumentUsageError(
+                    error_msg + " or endpoint account '--endpoint-account'."
+                )
```

Comment:
Not to mess with the logic too much here, but assuming if you have an identity, you don't need a connection string, but here we have no identity, AND no connection string:
Is it possible to add something about `Please provide a connection string OR a managed identity, otherwise...` or `No managed identity or connection string provided...` ? Or am I maybe overthinking it?